apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -99,6 +99,7 @@ export class RunJob {
           events: job.digest?.events,
           job,
           tags: notification.tags || [],
+          statelessPreferences: job.preferences,
```

Comment:
Is it necessary to call these `statelessPreferences`? We are storing them regardless of the type of the workflow, right?

Reply:
No, only stateless workflow executions have this stored on the `JobEntity`. Stateful Workflows have their preferences stored on the `PreferenceEntity` during sync, [here](https://github.com/novuhq/novu/blob/next/apps/api/src/app/bridge/usecases/sync/sync.usecase.ts#L219).

Reply:
Yes. Let me rephrase: would it be easier for the use case and our reasoning about it to store all preferences in the job when triggered? That way, we ensure that all types of workflows have the same preference resolution in case preferences change as the workflow executes.

Reply:
From a User's point of view, it's arguable that preference resolution should use the most up-to-date Preference to ensure that Workflows with long delays use the most recently modified Preference, rather than the Preference set at trigger time. If we store preferences on the job for stateful Workflow executions, we risk not respecting the most up to date Preference, and Subscriber's could receive notifications when the User didn't intend them to.

One inverted approach that @djabarovgeorge and I discussed to add consistent Preference resolution for both stateless + stateful Workflow executions involves storing a "transient" `NotificationTemplateEntity` & `Preference` entity for Stateless executions during Trigger, each of these entities can have a `stateful: boolean` flag added to indicate they are transient, to support API client lookups that only care about stateful entities. 
This approach will enable the Worker to be refactored to not care about stateless vs stateful entirely. There is considerable refactoring required to take this approach, however.

Reply:
I see. Let's part it for now although we need to get back to this.